mqtt/mqtt-broker/src/broker.rs

```diff
@@ -181,10 +184,10 @@ where
                 Ok(())
             }
             ClientEvent::PubAck0(id) => self.process_puback0(&client_id, id),
-            ClientEvent::PubAck(ref puback) => self.process_puback(&client_id, puback),
-            ClientEvent::PubRec(ref pubrec) => self.process_pubrec(&client_id, pubrec),
-            ClientEvent::PubRel(ref pubrel) => self.process_pubrel(&client_id, pubrel),
-            ClientEvent::PubComp(ref pubcomp) => self.process_pubcomp(&client_id, pubcomp),
+            ClientEvent::PubAck(puback) => self.process_puback(&client_id, &puback),
```

Comment:
nit: since you've touched this, why do we even need refs here? we are deconstructing the event anyways, why not move into `process_puback`. Just a question, no need to be fixed in this PR :)

Reply:
good catch!

Reply:
Actually it's a clippy that forces to borrow values that were not consumed:
https://rust-lang.github.io/rust-clippy/master/index.html#needless_pass_by_value